registry/message.go

```diff
@@ -78,5 +74,17 @@ func (r *Registry) analyseMessage(fileData *data.File, packageName, fileName str
 		r.analyseField(fileData, data, packageName, f)
 	}
 
+	fieldList := make([]string, 0)
+	// build field name list to store in typeInfo
+	for _, field := range data.Fields {
+		if !field.IsExternal && !field.IsRepeated && !field.IsOneOfField {
+			fieldList = append(fieldList, field.Name)
```

Comment:
wouldn't the `FieldList` here a little bit limited? wondering how would you handle nested type from external.

Reply:
Yes after giving a lot of thought, an implementation like this won't know the nature of the request payload during runtime. So I will probably be scraping this implementation...